internal/policy/verify.go

```diff
@@ -643,3 +649,46 @@ func getChangedPaths(repo *git.Repository, entry *rsl.ReferenceEntry) ([]string,
 
 	return gitinterface.GetDiffFilePaths(currentCommit, priorCommit)
 }
+
+func checkCommitAgainstModifiedPaths(ctx context.Context, repo *git.Repository, commit *object.Commit, commitPolicy *State) ([]string, error) {
+	filePaths, err := gitinterface.GetFilePathsChangedByCommit(repo, commit)
+	if err != nil {
+		return []string{}, err
+	}
+
+	badPaths := []string{}
+	for _, path := range filePaths {
+		keys, err := commitPolicy.FindPublicKeysForPath(ctx, fmt.Sprintf("file:%s", path))
+		if err != nil {
+			return []string{}, err
+		}
+
+		// Verify the keys associated with the commit. If verification fails, add the path to the list of bad paths.
+		err = verifyKeys(ctx, commit, keys)
```

Comment:
I appreciate using a helper but given we're consolidating this in #168, you can probably skip this.

Reply:
I have removed the helper function